sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -112,7 +113,11 @@ def __init__(self,
     self._state_handler_factory = GrpcStateHandlerFactory(
         self._state_cache, credentials)
     self._profiler_factory = profiler_factory
-    self._fns = {}  # type: Dict[str, beam_fn_api_pb2.ProcessBundleDescriptor]
+    self._fns = KeyedDefaultDict(
+        lambda id: self._control_stub.GetProcessBundleDescriptor(
+            beam_fn_api_pb2.GetProcessBundleDescriptorRequest(
```

Comment:
What appears in the default dict if GetProcessBundleDescriptor raises an error because the call fails?
Does it get propagated to the caller and it fails processing the bundle (is anything left in the map)?

Reply:
The error gets raised and nothing gets into the map.